gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java

```diff
@@ -239,20 +238,33 @@ public HistorianService getMetricsName(JsonObject params, Handler<AsyncResult<Js
         LOGGER.debug("max limit:" + max);
         query.setRows(max);
         query.addField(RESPONSE_METRIC_NAME_FIELD);
+        query.set("group", "true");
+        query.set("group.field", RESPONSE_METRIC_NAME_FIELD);
+        query.set("group.ngroups", "true");
+        query.set("group.limit", "-1");
         Handler<Promise<JsonObject>> getMetricsNameHandler = p -> {
             try {
                 final QueryResponse response = solrHistorianConf.client.query(solrHistorianConf.collection, query);
-                SolrDocumentList solrDocuments = response.getResults();
-                LOGGER.debug("Found " + response.getRequestUrl() + response + " result" + solrDocuments);
-                JsonArray metrics = new JsonArray(solrDocuments.stream()
-                        .map(SolrDocument::getFieldValuesMap)
-                        .map(stringCollectionMap -> stringCollectionMap.get(RESPONSE_METRIC_NAME_FIELD))
-                        .map(objects -> objects.toArray()[0].toString())
-                        .collect(Collectors.toList())
-                );
+                GroupResponse solrDocumentsGroup = response.getGroupResponse();
+                List<GroupCommand> results = solrDocumentsGroup.getValues();
+                JsonArray metrics = new JsonArray();
+                int totalMetricsFound = 0;
+                int totalNamesFound = 0;
+                for (GroupCommand gc : results) {
+                    totalMetricsFound = gc.getMatches();
+                    totalNamesFound = gc.getValues().size();
+                    List<Group> groups = gc.getValues();
+                    for(Group group : groups) {
+                        group.getGroupValue();
+                        LOGGER.debug(group.getGroupValue());
+                        metrics.add(group.getGroupValue());
+                        }
+                    }
+                LOGGER.debug("Found " + response.getRequestUrl() + response + " result : " + metrics);
                 LOGGER.debug("mertics :: "+ metrics);
                 p.complete(new JsonObject()
-                        .put(RESPONSE_TOTAL_FOUND, solrDocuments.getNumFound())
+                        .put(RESPONSE_TOTAL_METRICS_FOUND, totalMetricsFound)
```

Comment:
There is a problem here as discussed in chat. RESPONSE_TOTAL_METRICS_FOUND makes no sense here as it is the number of chunk with a name matching one of the returned name.

Reply:
i think this is an old version of code